src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -52,17 +49,18 @@ public void addVertex(VariableIdentifier vID)
   }
   public void addVertex(ExpressionVertex v) {
     nonVariableVertices.add(v);
+    allVertices.add(v);
   }
 
-  public List<ExpressionVertex> getVertices() {
-    List<ExpressionVertex> vertices = new LinkedList<>();
-    vertices.addAll(getVariableVertices().values());
-    vertices.addAll(getNonVariableVertices());
-    return vertices;
+  public Collection<ExpressionVertex> getVertices() {
+    return allVertices;
   }
 
   private List<ExpressionEdge> edges = new ArrayList<>();
   public void addEdge(ExpressionEdge e) {
+    Preconditions.checkArgument(getVertices().contains(e.getSource())
+                    && e.getTarget() == null || getVertices().contains(e.getTarget()),
+            "Edge had unexpected vertices " + e.toString());
```

Comment:
There are cases where the to vertex is null initially and added later (check the graph builder code).

As for the indentation, both were 1 level deeper than they should've been, not sure why. The 1st line checks the source, and the 2nd line checks the target. The 1st line could be on its own line but that seems unnecessary and increases vertical space used...

Also that 2nd line you wrote is 112 characters long :stuck_out_tongue_closed_eyes: